vnext/Scripts/IntegrationTests.ps1

```diff
@@ -1,97 +1,94 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-param(
-	[switch] $NoSetup,
-	[switch] $Run,
-	[switch] $Cleanup,
+#
+# IntegrationTests.ps1
+#
+param (
+	[switch] $NoServers,
+
 	[ValidateSet('x64', 'x86')]
 	[string] $Platform = 'x64',
+
 	[ValidateSet('Debug', 'Release')]
 	[string] $Configuration = 'Debug',
+
 	[string[]] $Tests,
-	$Assembly = "$PSScriptRoot\..\target\$Platform\$Configuration\React.Windows.Desktop.IntegrationTests\React.Windows.Desktop.IntegrationTests.dll",
-	$VsTest = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe",
+
+	[switch] $Preload,
+
 	[UInt32] $Delay = 10,
+
+	[ValidateScript({Test-Path $_})]
+	[string[]] $Assemblies =
+	(
+		"$PSScriptRoot\..\target\$Platform\$Configuration\" +
+		"React.Windows.Desktop.IntegrationTests\React.Windows.Desktop.IntegrationTests.dll"
+	),
+
+	[switch] $NoRun,
+
 	[switch] $List,
-	[switch] $Preload,
-	[string] $ReactNativeHome = "$($PSScriptRoot | Split-Path)\node_modules\react-native"
-)
 
-function Cleanup() {
-	# Expected netstat header:
-	# P1 P2     P3                     P4                     P5              P6
-	#    Proto  Local Address          Foreign Address        State           PID
-	try {
-		$process = (netstat -ano | Select-String ':8081')[0] | ConvertFrom-String
-		(Get-Process -Id $process.p6).Kill()
-	}
-	catch {
-		Write-Host 'Metro bundler process not found.'
-	}
+	[ValidateScript({Test-Path $_})]
+	[string] $VsTest =	"${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise\" +
+						"Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe",
 
-	try {
-		$process = (netstat -ano | Select-String ':5555')[0] | ConvertFrom-String
-		(Get-Process -Id $process.p6).Kill()
-	}
-	catch {
-		Write-Host 'WebSocket server process not found.'
-	}
-}
+	[ValidateScript({Test-Path $_})]
+	[string] $ReactNativeLocation = "$($PSScriptRoot | Split-Path)\node_modules\react-native",
+
+	[ValidateScript({Test-Path $_})]
+	[string] $NodePath = (Get-Command node.exe).Definition,
+
+	[ValidateScript({Test-Path $_})]
+	[string] $NpmPath = (Get-Command npm.cmd).Definition
+)
 
-#TODO: Cleanup Watchman logs, free ports 5555 and 8081.
+# Import test server functions.
+. $PSScriptRoot\TestServers.ps1
 
-$procs = @()
+if ($List) {
+	# https://github.com/Microsoft/vstest/issues/1732: Can't print full test names directly to console.
+	& $VsTest ($Assemblies -join ' ') --ListFullyQualifiedTests `
```

Comment:
>($Assemblies -join ' ')  [](start = 11, length = 24)

OOC, isn't that the default for passing array args?

Reply:
No. This expression is converting the string array to a single string separated by spaces.

Reply:
I understand that, I was just assuming that vstest does not expect a single string argument with multiple assembly names. It should be able to cope multiple string args, no?

---
In reply to: [304694068](https://github.com/microsoft/react-native-windows/pull/2768#discussion_r304694068) [](ancestors = 304694068)